src/main/scala/io/opentargets/etl/backend/openfda/stage/PrepareDrugList.scala

```diff
@@ -0,0 +1,32 @@
+package io.opentargets.etl.backend.openfda.stage
+
+import io.opentargets.etl.backend.ETLSessionContext
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.functions.{array, array_distinct, col, explode, flatten, lower}
+
+/*
+    Project     : io-opentargets-etl-backend
+    Timestamp   : 2021-08-04T11:44
+    Author      : Manuel Bernal Llinares <mbdebian@gmail.com>
+*/
+
+object PrepareDrugList {
+  def apply(dfChembl: DataFrame)(implicit context: ETLSessionContext) = {
+    val drugList = dfChembl
+      .selectExpr("id as chembl_id",
+        "synonyms as synonyms",
+        "name as pref_name",
+        "tradeNames as trade_names")
+      .withColumn("drug_names",
+        array_distinct(
+          flatten(array(col("trade_names"), array(col("pref_name")), col("synonyms")))))
+      .withColumn("_drug_name", explode(col("drug_names")))
+      .withColumn("drug_name", lower(col("_drug_name")))
+      .select("chembl_id", "drug_name")
+      .distinct()
+      .orderBy(col("drug_name"))
+
+    // NOTE - I can probably remove this and leave just the operation on dfChembl?
```

Comment:
You can remove it, but it's often useful to debugging to have it structured like this as you can set a breakpoint on.